cpp/tests/io/parquet_test.cpp

```diff
@@ -4220,10 +4220,14 @@ TEST_F(ParquetWriterTest, ByteArrayStats)
   std::vector<uint8_t> expected_col1_min{0xfe, 0xfe, 0xfe};
   std::vector<uint8_t> expected_col1_max{0xfe, 0xfe, 0xfe};
 
-  cudf::test::lists_column_wrapper<uint8_t> list_int_col0{
-    {0xf0}, {0xf0, 0xf5, 0xf3}, {0xf0, 0xf5, 0xf5}};
-  cudf::test::lists_column_wrapper<uint8_t> list_int_col1{
-    {0xfe, 0xfe, 0xfe}, {0xfe, 0xfe, 0xfe}, {0xfe, 0xfe, 0xfe}};
+  cudf::test::lists_column_wrapper<int8_t> list_int_col0{
+    {static_cast<int8_t>(0xf0)},
+    {static_cast<int8_t>(0xf0), static_cast<int8_t>(0xf5), static_cast<int8_t>(0xf3)},
+    {static_cast<int8_t>(0xf0), static_cast<int8_t>(0xf5), static_cast<int8_t>(0xf5)}};
+  cudf::test::lists_column_wrapper<int8_t> list_int_col1{
+    {static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe)},
+    {static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe)},
+    {static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe), static_cast<int8_t>(0xfe)}};
```

Comment:
I don't think we will need `static_cast`. In case of potential over/underflow, we should use `reinterpret_cast`.

Reply:
@ttnghia Can you explain why you think that `reinterpret_cast` is the correct choice here? We are relying on UB in the `static_cast`, but I don't think `reinterpret_cast` makes this better (relies on UB for type aliasing instead of UB for overflow). I don't see this as a case listed in "Only the following conversions can be done with `reinterpret_cast`: https://en.cppreference.com/w/cpp/language/reinterpret_cast

@hyperbolic2346 I think this is an example that adds to the evidence that we may want to use uint8 instead of int8.

Reply:
`static_cast` may do the conversion which may incur rounding off if the input value are out of bound of the target type.

`reinterpret_cast` on the other hand will not do anything. It just treats the input value as it is of the target type so there will not be any rounding off.

Reply:
Since we want the input values to be copied exactly into memory without data corruption, `reinterpret_cast` should be the right tool.

Reply:
@ttnghia The suggestion of `reinterpret_cast` doesn't compile. https://godbolt.org/z/G7r6a5x86

`reinterpret_cast` is typically only used for pointers/references. See also: https://stackoverflow.com/questions/18544849/how-to-use-reinterpret-cast-in-c

Reply:
Oh no.....
So there would be another cleaner way than using multiple `static_cast`: construct a `lists_column_wrapper<uint8_t>` like previously, then call `cudf::bit_cast` to `lists_column_wrapper<int8_t>`.

Reply:
But keeping the current `static_cast` way is not a big deal anyway :smile:

Reply:
I believe this turned into a non-issue. Feel free to bring it back if it isn't resolved with the latest code.